R/utils.R

```diff
@@ -216,3 +216,11 @@ set_lang <- function(new_lang) {
 reset_lang <- function(old_lang) {
   if (is.na(old_lang)) Sys.unsetenv("LANGUAGE") else Sys.setenv(LANGUAGE = old_lang)
 }
+
+#' Create a \code{linter} closure
+#' @param fun A function that takes a source file and returns \code{lint} objects.
+#' @return The same function with its class set to 'linter'.
+#' @export
+Linter <- function(fun) { # nolint: object_name_linter.
```

Comment:
Since we're exporting, I'd add `stopifnot(is.function(fun))` here too

Reply:
Adding that and that `fun` must have exactly one argument.
We can re-use that in `lint.R` for the function case (`obj <- Linter(obj)` will then perform the check over there)